packages/lib/next-auth/auth-options.ts

```diff
@@ -138,6 +139,22 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
         merged.emailVerified = user.emailVerified?.toISOString() ?? null;
       }
 
+      if ((trigger === 'signIn' || trigger === 'signUp') && account?.provider === 'google') {
```

Comment:
It's not optimal to run this every sign up or sign in via Google, but we don't have access to `identityProvider` to see if it's correctly set.

Reply:
We could easily skip out if we know the user is already verified no?

Reply:
It would, but we don't know if the provider is correctly set on the user table

Reply:
Slap it in the token while we're here, this saves tripping to the db all the time 😄

Reply:
Actually don't worry about it, we trip enough as is and this is only on signIn or signUp